ratis-examples/src/main/java/org/apache/ratis/examples/filestore/cli/DataStream.java

```diff
@@ -118,9 +121,11 @@ protected void operation(RaftClient client) throws IOException {
     FileStoreClient fileStoreClient = new FileStoreClient(client);
     System.out.println("Starting DataStream write now ");
 
+    final ExecutorService executor = Executors.newFixedThreadPool(getNumFiles());
```

Comment:
It should have a max since getNumFiles() can return a large number.

Reply:
BTW, we should make the executors in FileStore configurable.  I suspect the performance will be better if we use more threads.
```
//org.apache.ratis.examples.filestore.FileStore
  private final ExecutorService writer = Executors.newFixedThreadPool(10);
  private final ExecutorService committer = Executors.newFixedThreadPool(3);
  private final ExecutorService reader = Executors.newFixedThreadPool(10);
  private final ExecutorService deleter = Executors.newFixedThreadPool(3);
```

Reply:
@szetszwo Sorry, what the following three thread pool used for ?  
private final ExecutorService committer = Executors.newFixedThreadPool(3);
  private final ExecutorService reader = Executors.newFixedThreadPool(10);
  private final ExecutorService deleter = Executors.newFixedThreadPool(3);

Reply:
committer: committing files in applyTransaction
reader: for read requests
deleter: for delete requests

Reply:
why delete and read request related to this PR ? Because we test the performance of write.

Reply:
@szetszwo I see.

Reply:
We may do the server side configurable feature separately.